booster/library/Booster/Pattern/Rewrite.hs

```diff
@@ -284,16 +284,16 @@ applyRule pat@Pattern{ceilConditions} rule = withRuleContext rule $ runRewriteRu
     -- unify terms
     subst <- withContext "match" $ case matchTerms Rewrite def rule.lhs pat.term of
         MatchFailed (SubsortingError sortError) -> do
-            withContext "abort" $ logPretty sortError
+            withContext "error" $ logPretty sortError
             failRewrite $ RewriteSortError rule pat.term sortError
         MatchFailed err@ArgLengthsDiffer{} -> do
-            withContext "abort" $ logPretty err
+            withContext "error" $ logPretty err
             failRewrite $ InternalMatchError $ renderText $ pretty err
         MatchFailed reason -> do
             withContext "failure" $ logPretty reason
             fail "Rule matching failed"
         MatchIndeterminate remainder -> do
-            withContext "abort" $
+            withContext "indeterminate" $
                 logMessage $
                     WithJsonMessage (object ["remainder" .= (bimap externaliseTerm externaliseTerm <$> remainder)]) $
                         renderOneLineText $
```

Comment:
We need to add these new contexts to the docs to keep them up to date, especially as we now have (off the top of my head):

* match,error
* match,indeterminate
* definedness,abort
* condition,abort
* failure
* failure,break
* failure,continue

Would it make sense to make abort always signify that we fall back? so that instead of `match,error` we could have something like `match,internal-error,abort` and `match,indeterminate` would be `match,indeterminate,abort` ? Or would this break the filtering?

Reply:
I agree we should try to keep a complete list of used context strings - or better, use a type, as we discussed yesterday. I did not start implementing type-based contexts just yet because I just wanted to get things into a state that gives us the same data as before first.

The `condition` (agree, let's rename to `constraint`) and `definedness` contexts were added to distinguish from the context alone what the reason was.

The `error` logs above are something we never want to see, actual server errors IMHO (inconsistent terms with the definition, despite having been internalised).

The case of `match,abort.` (now `match,indeterminate.`) is special because in `performRewrite` we have a mechanism to simplify (with booster only) and re-try the rewrite, which will often succeed. The log entry that is actually counted (`rewrite*,match,abort.`) is emitted there when we know we will indeed abort the rewrite.

Reply:
I have added the new contexts to the documentation and adapted the explanations there.